Add DocxTemplaterBulk

ortizman opened the pull request:
With this template engine, the library can process docx with many more scripts
Fix #21

Commit:
Add DocxTemplaterBulk. With this template engine, the library can process docx with many more scripts

ortizman commented:
@snowindy

snowindy commented:
Thank you for the contribution, I will check it and try to merge if it's alright!

snowindy commented:
Hi!
Sorry for being late! Could you please give me an example document that gives "Method code too large!" error?

I think that manually adding BREAK tag somewhere in a document is a hack and it gives bad user experience. That is, if you are new library user, you won't know this, and just think it's malfunctioning.

I think I can invent a way to avoid error "Method code too large!" without it, with your example to reproduce.

Commit:
Add test that reproduces " Method code too large! "
DocxTemplateBulk success test
DocxTemplate fail test (expected runtimeexception)

ortizman commented:
> Hi!
> Sorry for being late! Could you please give me an example document that gives "Method code too large!" error?
> 
> I think that manually adding BREAK tag somewhere in a document is a hack and it gives bad user experience. That is, if you are new library user, you won't know this, and just think it's malfunctioning.
> 
> I think I can invent a way to avoid error "Method code too large!" without it, with your example to reproduce.

@snowindy 
I created a unity test to reproduce the limitation. DocxTemplateBulk passes it and DocxTemplate does not (expected=RuntimeException.class)
I really tried not to add the BREAK. But I can't imagine an alternative that doesn't require breaking the document and processing it in parts, since it is a limitation of the JVM.
https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.9.1

An alternative would be to process the document in parts in a way that is transparent to the client of the library. But this leads to the problem of knowing where to make the cut without breaking any scriptlet.

At the moment this hack is being used productively in several documents, many larger than the test file, without drawbacks
